test/binaryplatforms.jl

```diff
@@ -330,8 +330,7 @@ end
     # Ambiguity test
     @test select_platform(platforms, P("aarch64", "linux")) == "linux3"
     @test select_platform(platforms, P("aarch64", "linux"; libgfortran_version=v"3")) == "linux3"
-    # This one may be surprising, but we still match `linux3`, and since linux3 is shorter, we choose it.
-    @test select_platform(platforms, P("aarch64", "linux"; libgfortran_version=v"3", libstdcxx_version=v"3.4.18")) === "linux3"
+    @test select_platform(platforms, P("aarch64", "linux"; libgfortran_version=v"3", libstdcxx_version=v"3.4.18")) === "linux5"
```

Comment:
As an aside, this "surprise" should have been my hint that my previous implementation was incorrect.